apps/portals/src/AppInitializer.tsx

```diff
@@ -97,16 +96,16 @@ function useSession(
       console.error('Error on getSession: ', e)
       // intentionally calling sign out because there token could be stale so we want
       // the stored session to be cleared out.
-      SynapseClient.signOut(() => {
-        // PORTALS-2293: if the token was invalid (caused an error), reload the app to ensure all children
-        // are loading as the anonymous user
-        window.location.reload()
-      })
+      await SynapseClient.signOut()
+      // PORTALS-2293: if the token was invalid (caused an error), reload the app to ensure all children
+      // are loading as the anonymous user
+      window.location.reload()
     }
   }, [initAnonymousUserState])
 
-  const resetSession = useCallback(() => {
-    SynapseClient.signOut(getSession)
+  const resetSession = useCallback(async () => {
+    await SynapseClient.signOut()
+    await getSession()
```

Comment:
No reason to pass a callback argument to `SynapseClient.signOut` when we can easily do
`await SynapseClient.signOut()` or `signOut().then(...)`. IMO the function argument just makes it easier to accidentally write a race condition.